### ISSUE TITLE
[Bug]: Market Feedback Agent keeps stopping/Android app freezing

### ISSUE THREAD

**Issue by Duvenel** [MelvorIdle/melvoridle.github.io#2806]
### Describe the bug

As soon as a save is loaded into the game will lag on open. When trying to interact with any UI element the game will struggle before finally loading and after a few seconds an error message will pop up and give 3 options: 'ok', 'report' and 'wait'. 

Clicking 'ok', closes the game and trying to reopen gives the same results. Clicking 'wait' gives a few seconds of playability before the game hangs again.

I've tried clearing the cache, and app data separately as well as uninstalling and reinstalling and the issue persists. I've also disabled all mods and the issue is still there. 

The issue seemed to happen not long after I bought the Throne of the Herald dlc.

### Reproduction Steps

1. Load into any save
2. Wait a few seconds or try to navigate with the UI
3. Wait for the game to freeze and error popup displays

### Expected behaviour

Loading into any save causes an error popup to display. Hitting 'ok' force closes the game. Hitting 'wait' allows for what seems to be normal functionality for a few seconds before the same error pops up.

### Save Link

https://melvoridle.com/save/zNIlc4HC

### Screenshots

![Screenshot_2022-10-25-20-10-03-195_com google android feedback](https://user-images.githubusercontent.com/8366852/197864272-868b374e-a431-486e-990b-249a0479cda7.jpg)
![Screenshot_2022-10-25-20-10-35-285_com miui bugreport](https://user-images.githubusercontent.com/8366852/197864279-e5238c22-aee3-4b3d-8c7c-d1701580b729.jpg)
![Screenshot_2022-10-25-20-10-44-120_com miui bugreport](https://user-images.githubusercontent.com/8366852/197864287-0cd83f5d-7e74-46dd-ba67-d82145d84c2f.jpg)
![Screenshot_2022-10-25-20-10-48-151_com miui bugreport](https://user-images.githubusercontent.com/8366852/197864288-c95a8cf9-3716-4608-9f0d-ab13672d5254.jpg)

### Console output

java.lang.RuntimeException: Unable to start activity ComponentInfo{com.google.android.feedback/com.google.android.feedback.FeedbackActivity}: java.lang.SecurityException: getDataNetworkTypeForSubscriber
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3740)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3907)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2260)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:210)
	at android.os.Looper.loop(Looper.java:299)
	at android.app.ActivityThread.main(ActivityThread.java:8168)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:556)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1037)
Caused by: java.lang.SecurityException: getDataNetworkTypeForSubscriber
	at android.os.Parcel.createExceptionOrNull(Parcel.java:2425)
	at android.os.Parcel.createException(Parcel.java:2409)
	at android.os.Parcel.readException(Parcel.java:2392)
	at android.os.Parcel.readException(Parcel.java:2334)
	at com.android.internal.telephony.ITelephony$Stub$Proxy.getNetworkTypeForSubscriber(ITelephony.java:9300)
	at android.telephony.TelephonyManager.getNetworkType(TelephonyManager.java:2964)
	at android.telephony.TelephonyManager.getNetworkType(TelephonyManager.java:2928)
	at com.google.android.feedback.FeedbackSession.getTelephonyInfo(FeedbackSession.java:307)
	at com.google.android.feedback.FeedbackSession.<init>(FeedbackSession.java:152)
	at com.google.android.feedback.FeedbackSession.<init>(FeedbackSession.java:141)
	at com.google.android.feedback.FeedbackActivity.pushSession(FeedbackActivity.java:209)
	at com.google.android.feedback.FeedbackActivity.createFeedbackSession(FeedbackActivity.java:178)
	at com.google.android.feedback.FeedbackActivity.onCreate(FeedbackActivity.java:96)
	at android.app.Activity.performCreate(Activity.java:8178)
	at android.app.Activity.performCreate(Activity.java:8146)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1330)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3713)
	... 12 more

### Which platforms are you experiencing this bug on?

- [ ] Chrome
- [ ] Firefox
- [ ] Edge
- [ ] Safari
- [ ] Mobile App (iOS)
- [X] Mobile App (Android)
- [ ] Steam
- [ ] Other (Please Specify)

### Which version of the game are you experiencing this bug on?

v1.1.0

### Game Subversion

?4615

### Are you using any mods?

No

**Comment by Mot92** [comment]
Similar problem, after the release of the game on October 20, it worked without problems. A week later, it began to fly out as written by the author of the topic. I am attaching the error code.
![Screenshot_2022-10-29-18-13-07-862_com miui bugreport](https://user-images.githubusercontent.com/57660738/198836442-b9b64738-cad2-4a7c-8212-94e5f41d13cd.jpg)
![Screenshot_2022-10-29-18-13-12-381_com miui bugreport](https://user-images.githubusercontent.com/57660738/198836447-387c4a3b-4ac5-4fa1-a65a-617ec56ae15f.jpg)


Console output
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.google.android.feedback/com.google.android.feedback.FeedbackActivity}: java.lang.SecurityException: getDataNetworkTypeForSubscriber
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3761)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3935)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2286)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:210)
	at android.os.Looper.loop(Looper.java:299)
	at android.app.ActivityThread.main(ActivityThread.java:8292)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:556)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1045)
Caused by: java.lang.SecurityException: getDataNetworkTypeForSubscriber
	at android.os.Parcel.createExceptionOrNull(Parcel.java:2425)
	at android.os.Parcel.createException(Parcel.java:2409)
	at android.os.Parcel.readException(Parcel.java:2392)
	at android.os.Parcel.readException(Parcel.java:2334)
	at com.android.internal.telephony.ITelephony$Stub$Proxy.getNetworkTypeForSubscriber(ITelephony.java:9300)
	at android.telephony.TelephonyManager.getNetworkType(TelephonyManager.java:2971)
	at android.telephony.TelephonyManager.getNetworkType(TelephonyManager.java:2935)
	at com.google.android.feedback.FeedbackSession.getTelephonyInfo(FeedbackSession.java:307)
	at com.google.android.feedback.FeedbackSession.<init>(FeedbackSession.java:152)
	at com.google.android.feedback.FeedbackSession.<init>(FeedbackSession.java:141)
	at com.google.android.feedback.FeedbackActivity.pushSession(FeedbackActivity.java:209)
	at com.google.android.feedback.FeedbackActivity.createFeedbackSession(FeedbackActivity.java:178)
	at com.google.android.feedback.FeedbackActivity.onCreate(FeedbackActivity.java:96)
	at android.app.Activity.performCreate(Activity.java:8241)
	at android.app.Activity.performCreate(Activity.java:8207)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1329)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3734)
	... 12 more

**Comment by Duvenel** [comment]
Update: Version 4641

I thought to try the game on my mobile's browser(Chrome) and the issue seems to persist to there as well. I've also tried to clear the cache for Chrome and this seems to have had some success with my mobile's browser version although the issue is still there on the app.

When I opened the app to test if clearing Chrome's cache would work it immediately started to freeze/hang the game again. When I then went back to my browser version the game had the same issue, so I cleared the cache for Chrome again and tried the browser once more. Chrome gave me another error but hitting 'wait' this time seemed to be fine and the game seems to run as normal in my mobile browser now.

I'm afraid to test on the app version again in case this breaks the functionality in my mobile browser because I can at least now play. This issue is extrememly frustrating and makes the game unplayable, it makes me wish I never bought the DLC as my game seems to have broken ever since I did.

I'll update further if I find anything else.
